imports/ui/case-explorer/case-explorer.jsx

```diff
@@ -136,100 +134,38 @@ class CaseExplorer extends Component {
-                          { unitCases.length } cases
-                        </div>
-                        {bzId && (
-                          <div className='no-shrink flex items-center'>
-                            <Link
-                              className='f6 link ellipsis ml3 pl1 mv1 bondi-blue fw5'
-                              to={`/case/new?unit=${bzId}`}>
-                              Add case
-                            </Link>
-                          </div>
-                        )}
-                      </div>
-                    </div>
-                  </div>
-                  {isExpanded && (
-                    <ul className='list bg-light-gray ma0 pl0 shadow-in-top-1'>
-                      <CaseList
-                        allCases={unitCases}
-                        onItemClick={() => dispatch(storeBreadcrumb(match.url))}
-                      />
-                    </ul>
-                  )}
-                </div>
+                <ItemRow unitTitle={unitTitle}
+                  expandedUnits={expandedUnits}
+                  items={unitCases}
+                  unitType={unitType}
+                  unitItems={unitCases}
+                  handleExpandUnit={this.handleExpandUnit}
+                  bzId={bzId}
+                  dispatch={dispatch}
+                  match={match}
+                />
               )
             })
-            : (
-              <div className='flex-grow flex flex-column items-center justify-center'>
-                <div className='tc'>
-                  <div className='dib relative'>
-                    <FontIcon className='material-icons' color='var(--moon-gray)' style={{fontSize: '6rem'}}>
-                      card_travel
-                    </FontIcon>
-                    <div className='absolute bottom--1 right--1 pb2'>
-                      <div className='br-100 pa1 bg-very-light-gray lh-cram'>
-                        <FontIcon className='material-icons' color='var(--moon-gray)' style={{fontSize: '2.5rem'}}>
-                          add_circle
-                        </FontIcon>
-                      </div>
-                    </div>
-                  </div>
-                  <div className='mt3 ph4'>
-                    <div className='mid-gray b lh-copy'>
-                      <div>There are no cases that match your current filter combination.</div>
-                      <div>Click on the "+" button to select a unit to add a new case</div>
-                    </div>
-                  </div>
-                </div>
-              </div>
-            )
+            : (<NoItemMsg item={'case'} />)
           }
         </div>
-        <div className='absolute right-1 bottom-2'>
-          <FloatingActionButton onClick={() => dispatch(push('/unit'))}>
-            <FontIcon className='material-icons'>add</FontIcon>
-          </FloatingActionButton>
-        </div>
+        <UnitBtn dispatch={dispatch} />
```

Comment:
Handing off "dispatch" to child component is a kind of an anti-pattern of Redux. You should either use a handler function instead, or make "UnitBtn" a "smart" component by wrapping its defnition with Redux's "connect()". I think in this context a handler function will make more sense though.